specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+                        // For instance, retrieve the updated memory usage information 
+                        // for that iframe.  
+                    }
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        //! [ICoreWebView2Frame:add_RendererProcessIdChanged]
+        frame5->add_RendererProcessIdChanged(
+            Callback<ICoreWebView2FrameRendererProcessIdChangedEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    IUnknown* args) -> HRESULT 
+                {
+                    wil::com_ptr<ICoreWebView2Frame5> frame5;
+                    sender->QueryInterface(IID_PPV_ARGS(&frame5));
+                    if (frame5) 
+                    {
+                        //! [ICoreWebView2Frame:get_RendererProcessId]
+                        UINT32 processId;
+                        CHECK_FAILURE(frame5->get_RendererProcessId(&processId));
+                        // Do something when renderer process id has been changed. 
+                        // For instance, retrieve the updated memory usage information
+                        // for that iframe. 
+                    }
+                    return S_OK;
+                    
+                })
+                .Get(),
+            NULL);
+    }
+```
+C#
+```c#
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    if (e.IsSuccess)
+    {
+        // ...
+        webView.CoreWebView2.FrameCreated += WebView_HandleIFrames;
+        // </RendererProcessIdChanged>
+        webView.CoreWebView2.RendererProcessIdChanged += WebView_RendererProcessIdChanged;
+        // ...
+    }
+}
+void WebView_HandleIFrames(object sender, CoreWebView2FrameCreatedEventArgs args)
+{
+    _webViewFrames.Add(args.Frame);
+    args.Frame.FrameCreated += WebViewFrames_CreatedNestedIFrames;
+    // </RendererProcessIdChanged>
+    args.Frame.RendererProcessIdChanged += WebViewFrames_RendererProcessIdChanged;
+    // ...
+}
+
+void WebViewFrames_CreatedNestedIFrames(object sender, CoreWebView2FrameCreatedEventArgs args)
+{
+    // Add frame created event for webViewFrames
+    _webViewFrames.Add(args.Frame);
+    // </FrameCreated>
+    args.Frame.FrameCreated += WebViewFrames_CreatedNestedIFrames;
+    // </RendererProcessIdChanged>
+    args.Frame.RendererProcessIdChanged += WebViewFrames_RendererProcessIdChanged;
+    // ...
+}
+
+void WebView_RendererProcessIdChanged(object sender, object args) 
+{
+    // </RendererProcessId>
+    var processId = webView.CoreWebView2.RendererProcessId;
+    // Do something when renderer process id has been changed. 
+    // For instance, retrieve the updated memory usage information 
+    // for that main frame. 
+}
+void WebViewFrames_RendererProcessIdChanged(object sender, object args)
+{
+    bool isWebviewFrame = (sender is CoreWebView2Frame);
+    if (isWebviewFrame) 

```

Comment:
Under what circumstances would this ever be false?

Reply:
With ProcessInfos change this is going away.